### PR TITLE
[FIX] account: properly handle move without sequence_number

### PR DESCRIPTION
**Pull request by yelizariev**
Since Odoo v15, user can rename invoice to a custom value. Odoo automatically
detects `sequence_number` for the new name. If no number is detected, the
`sequence_number` is set to zero. In this edge case, method
`_is_end_of_seq_chain` doesn't work and raises an exception, because `last_rec` is
empty, while `_get_last_sequence` has constrain `self.ensure_one()`.

Fix it by allowing `last_rec` with zero `sequence_number`.

STEPS:

* Create invoice
* Confirm
* Reset to Draft
* Rename invoice to "Test"
* Save
* Try to delete the invoice

BEFORE: ensure_one error

Once it's fixed, we face another problem: User Error *You cannot delete this entry, as it
has already consumed a sequence number and is not the last one in the chain. You
should probably revert it instead.*

It happens because `sequence_prefix` is computed as empty string, while
`_get_last_sequence` method ignores empty values. Fix it by allowing search by
empty string too.

AFTER the two changes user can delete draft invoice

opw-2861841

### CODE REVIEW

File: addons/account/models/sequence_mixin.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def _is_end_of_seq_chain(self):
             seq = format_values.pop('seq')
             batch = batched[(format, frozendict(format_values))]
             batch['seq_list'].append(seq)
-            if batch['last_rec'].sequence_number < record.sequence_number:
+            if not batch['last_rec'] or batch['last_rec'].sequence_number < record.sequence_number:
```
Comment (william-andre) [inline addons/account/models/sequence_mixin.py:271]: But you want to allow unlinking the move, don't you? You are still raising an error...
Doesn't this work?
```suggestion
            if batch['last_rec'].sequence_number <= record.sequence_number:
```

Comment (yelizariev) [reply addons/account/models/sequence_mixin.py:271]: I'm not sure. The invoice has sequence number equal to 0, which is not the last one. So it makes sense to block deleting, no?

Comment (william-andre) [reply addons/account/models/sequence_mixin.py:271]: It is the last one, otherwise it would not be 0

Comment (yelizariev) [reply addons/account/models/sequence_mixin.py:276]: Fixed. It happened because `sequence_prefix` was computed as empty string. Please review again

**Comment by william-andre** [comment]
@yelizariev please review the fixup commit

**Comment by yelizariev** [comment]
@william-andre 

I've checked it and updated commit message

<details>

```

[[FIX] account: properly handle move without sequence_number](https://github.com/odoo/odoo/pull/92876/commits/f4e5208e1a6cc0dea2e847d4905a146b04207850) 
[f4e5208](https://github.com/odoo/odoo/pull/92876/commits/f4e5208e1a6cc0dea2e847d4905a146b04207850)
Since Odoo v15, user can rename invoice to a custom value. Odoo automatically
detects `sequence_number` for the new name. If no number is detected, the
`sequence_number` is set to zero. In this edge case, method
`_is_end_of_seq_chain` doesn't work and raises an exception, because `last_rec` is
empty, while `_get_last_sequence` has constrain `self.ensure_one()`.

Fix it by allowing `last_rec` with zero `sequence_number`.

STEPS:

* Create invoice
* Confirm
* Reset to Draft
* Rename invoice to "Test"
* Save
* Try to delete the invoice

BEFORE: ensure_one error

Once it's fixed, we face another problem: User Error *You cannot delete this entry, as it
has already consumed a sequence number and is not the last one in the chain. You
should probably revert it instead.*

It happens because `sequence_prefix` is computed as empty string, while
`_get_last_sequence` method ignores empty values. Fix it by allowing search by
empty string too.

AFTER the two changes user can delete draft invoice

opw-2861841
```

</details>

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**